sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -141,6 +141,37 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
         writeOptions = c.options.filterKeys(_ != "path"),
         ignoreIfExists = c.ifNotExists)
 
+    case c @ CreateTableLikeStatement(target, source, loc, ifNotExists) =>
+      def validateLocation(loc: Option[String]) = {
+        if (loc.isDefined) {
+          throw new AnalysisException("Location clause not supported for " +
+            "CREATE TABLE LIKE statement when tables are of V2 type")
+        }
+      }
+      (target, source) match {
+        case (NonSessionCatalog(tCatalog, t), NonSessionCatalog(sCatalog, s)) =>
+          validateLocation(loc)
+          CreateTableLike(tCatalog.asTableCatalog,
+            t,
+            Some(sCatalog.asTableCatalog),
+            s,
+            ifNotExists)
+        case (NonSessionCatalog(tCatalog, t), SessionCatalog(sCatalog, s)) =>
```

Comment:
If we need to catch session catalog, we should move the case to `ResolveSessionCatalog`. It's OK to create v2 command in `ResolveSessionCatalog`

Reply:
@cloud-fan NonSessionCatalog is not available in ResolveSessionCatalog, right ? this case catches both NonSessionCatalog and SessionCatalog ?

Reply:
ok let's leave it.